### PR TITLE
feat: add utm_source channel mapping for personio

### PR DESCRIPTION
**Pull request by milljoniaer**
### What is included?

This PR allows us to map certain `utm_source` values to hardcoded personio Channels.
This way we can "track" where the applicants came from.

In this PR the first UTM_Source for Google Jobs is added.

This covers a suggestion made in #506

### CODE REVIEW

File: src/components/pages/career-details/career-details.tsx [hunk 1]
```diff
@@ -16,6 +16,10 @@ interface CareerDetailsProps {
 
 const DEFAULT_CHANNEL_ID = '329206';
 
+const UTM_CHANNEL_MAP = new Map<string | null, string | null>([
```
Comment (feedm3) [inline src/components/pages/career-details/career-details.tsx]: Looks overly complex. I would recommend to keep it a simple object:

```
const UTM_CHANNEL_MAP = {
  google_jobs_apply: '713571',
};
```

Also, add a comment above why we need to have this map.

File: src/components/pages/career-details/career-details.tsx [hunk 2]
```diff
@@ -60,7 +65,9 @@ export const CareerDetails = ({
 
       <Form
         company_id="41230"
-        recruiting_channel_id={personioChannel ?? DEFAULT_CHANNEL_ID}
+        recruiting_channel_id={
+          utmChannel ?? personioChannel ?? DEFAULT_CHANNEL_ID
```
Comment (feedm3) [inline src/components/pages/career-details/career-details.tsx]: To keep logic out of the JSX, I would also put this check into its own variable. This way, you have all related code to the `recruitingChannelId` in one place.

**Review by feedm3 — APPROVED** [review]
LGTM 👍 

I just have some smaller comments about refactoring, so I will already approve this PR.

File: src/components/pages/career-details/career-details.tsx (hunk 1 above)
Comment (milljoniaer) [reply src/components/pages/career-details/career-details.tsx]: Making it less complex seems cool, but if I make the map an Object I am not able to do this anymore:
image

I liked that I dont have to think about the `utm_source` value and can just take it as key for my map to get the correct channel_id. (even if this whole thing makes only sense with more channels than only google jobs)

Is there a less complex way to do this or shall I just leave out the whole mapping thing and something like:

```
googleChannel = searchParams.get("utm_source") === "google_jobs_apply" ? GOOGLE_CHANNEL : undefinded;
```

File: src/components/pages/career-details/career-details.tsx (hunk 2 above)
Comment (milljoniaer) [reply src/components/pages/career-details/career-details.tsx]: 👍

File: src/components/pages/career-details/career-details.tsx (hunk 1 above)
Comment (feedm3) [reply src/components/pages/career-details/career-details.tsx]: You can access an object with the array syntax (not sure if this is the official name 😄):

```
const utmChannel = UTM_CHANNEL_MAP[searchParams.get('utm_source')]
```

Comment (georgiee) [reply src/components/pages/career-details/career-details.tsx]: While you have a point with the types @milljoniaer I think we should go the much simpler way as outlined by @feedm3. We don't benefit from a map in this context and it introduces a lot of type noise. Make it an object and access it with the `bracket notation` (I had to search for it too @feedm3  😂, [reference](https://dev.to/capscode/dot-and-bracket-notation-in-javascript-object-12ij))

File: src/components/pages/career-details/career-details.tsx [hunk 3]
```diff
@@ -41,6 +45,9 @@ export const CareerDetails = ({
   const location = useLocation();
   const searchParams = new URLSearchParams(location.search);
   const personioChannel = searchParams.get('_pc');
+  const utmChannel = UTM_CHANNEL_MAP.get(searchParams.get('utm_source'));
+
+  const recruitingChannel = utmChannel ?? personioChannel ?? DEFAULT_CHANNEL_ID;
```
Comment (georgiee) [inline src/components/pages/career-details/career-details.tsx:50]: I like, that's a clear order  👍

**Review by georgiee — APPROVED** [review]
Lovely change 🙌 Let's change the map to an object.
Functionality is good already so I approve too 👍

File: src/components/pages/career-details/career-details.tsx (hunk 1 above)
Comment (milljoniaer) [reply src/components/pages/career-details/career-details.tsx]: Thank you for the bracket notation tipp, I did not knew about this, and this way it feels convenient again 👍

**Pull request merged**